Fix some JerryScript v2.1.0 build issues in our environment:
	- Get PATH_MAX from linux/limits.h
	- Use the more widely available strlen() at the loss of a small length
	  checking optimisation.
	- Don't use putchar/fputc as this can cause compile errors with this
	  project's sign-conversion warnings-as-errors on some platforms (e.g.
	  uClibc on x86_64.

--- jerryscript/jerry-port/default/default-module.c	2019-10-22 10:18:34.184989941 +1000
+++ jerryscript/jerry-port/default/default-module.c	2019-10-22 10:59:06.667120320 +1000
@@ -17,6 +17,7 @@
 #include <libgen.h>
 #endif /* !defined (WIN32) */
 #include <limits.h>
+#include <linux/limits.h>
 #include <stdarg.h>
 #include <stdlib.h>
 #include <string.h>
@@ -152,7 +153,7 @@
 
   if (norm_p != NULL)
   {
-    const size_t len = strnlen (norm_p, out_buf_size);
+    const size_t len = strlen (norm_p);
     if (len < out_buf_size)
     {
       strncpy (out_buf_p, norm_p, out_buf_size);
--- jerryscript/jerry-port/default/default-io.c	2019-10-22 10:13:50.000000000 +1000
+++ jerryscript/jerry-port/default/default-io.c	2019-10-22 14:35:44.843886243 +1000
@@ -116,7 +116,7 @@
 void
 jerry_port_print_char (char c) /**< the character to print */
 {
-  putchar (c);
+  fwrite(&c, sizeof c, 1, stdout);
 
 #if defined (JERRY_DEBUGGER) && (JERRY_DEBUGGER == 1)
   debug_buffer[debug_buffer_index++] = c;
